c10/core/WrapDimMinimal.cpp

```diff
@@ -0,0 +1,36 @@
+#include <c10/core/WrapDimMinimal.h>
+
+namespace c10 {
+namespace detail {
+
+int64_t maybe_wrap_dim_slow(
+    int64_t dim,
+    int64_t dim_post_expr,
+    bool wrap_scalar) {
+  if (dim_post_expr <= 0) {
```

Comment:
Very odd that this is `<= 0`. @ezyang is this intended? I would assume that `dim_post_expr` must be nonnegative.